pkg/types/vsphere/conversion/installconfig.go

```diff
@@ -71,38 +145,124 @@ func ConvertInstallConfig(config *types.InstallConfig) error {
-		return fmt.Sprintf("/%s/host/%s", datacenter, cluster)
+// SetComputeClusterPath returns the cluster path either by path.Join or using the function
+// findViaPathOrName.
+func SetComputeClusterPath(finder *find.Finder, cluster, datacenter string) (string, error) {
+	if cluster != "" && !path.IsAbs(cluster) {
+		clusterPathFind := path.Join("/", datacenter, "host", "...")
+		clusterPath := path.Join("/", datacenter, "host", cluster)
+
+		if finder == nil {
+			localLogger.Warnf("computeCluster as a non-path is now deprecated; please use the form: %s", clusterPath)
+			return clusterPath, nil
+		}
+
+		newClusterPath, err := findViaPathOrName(finder, clusterPath, clusterPathFind)
+		if err != nil {
+			return "", err
+		}
+
+		if clusterPath != newClusterPath {
+			localLogger.Debugf("cluster path changed from %s to %s", clusterPath, newClusterPath)
+		}
+		localLogger.Warnf("computeCluster as a non-path is now deprecated; please use the form: %s", newClusterPath)
+
+		return newClusterPath, nil
 	}
-	return cluster
+	return cluster, nil
 }
 
-func setDatastorePath(datastore, datacenter string) string {
-	if datastore != "" && !strings.HasPrefix(datastore, "/") {
-		localLogger.Warnf("datastore as a non-path is now deprecated; please use the form: /%s/datastore/%s", datacenter, datastore)
-		return fmt.Sprintf("/%s/datastore/%s", datacenter, datastore)
+// SetDatastorePath returns the datastore path either by path.Join or using the function
+// findViaPathOrName.
+func SetDatastorePath(finder *find.Finder, datastore, datacenter string) (string, error) {
+	if datastore != "" && !path.IsAbs(datastore) {
+		datastorePath := path.Join("/", datacenter, "datastore", datastore)
+		datastorePathFind := path.Join("/", datacenter, "datastore", "...")
+
+		// There is no connection to vCenter, use unconnected method for generating path
+		if finder == nil {
+			localLogger.Warnf("datastore as a non-path is now deprecated; please use the form: %s", datastorePath)
+			return datastorePath, nil
+		}
+
+		newDatastorePath, err := findViaPathOrName(finder, datastorePath, datastorePathFind)
+		if err != nil {
+			return "", err
+		}
+		if datastorePath != newDatastorePath {
+			localLogger.Debugf("datastore path changed from %s to %s", datastorePath, newDatastorePath)
+		}
+		localLogger.Warnf("datastore as a non-path is now deprecated; please use the form: %s", newDatastorePath)
+		return newDatastorePath, err
 	}
-	return datastore
+	return datastore, nil
 }
 
-func setFolderPath(folder, datacenter string) string {
-	if folder != "" && !strings.HasPrefix(folder, "/") {
-		localLogger.Warnf("folder as a non-path is now deprecated; please use the form: /%s/vm/%s", datacenter, folder)
-		return fmt.Sprintf("/%s/vm/%s", datacenter, folder)
+// SetFolderPath returns the folder path either by path.Join or using the function
+// findViaPathOrName.
+func SetFolderPath(finder *find.Finder, folder, datacenter string) (string, error) {
```

Comment:
I just realized all of these functions are identical except for one token used in the path and messages. I know "message composition" is Bad™... when translating. But we're not doing that, are we?

My point: Could these three funcs could be DRYed down to one?